### ISSUE TITLE
🐛 plugins[].includes is matched against the absolute file path, unlike files.includes

### ISSUE THREAD

**Issue by everton-dgn** [biomejs/biome#11082]
### Environment information

```block
CLI:
  Version:                      2.5.2

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:           npm
  BIOME_CONFIG_PATH:            unset
  JS_RUNTIME_VERSION:           v24.18.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/11.9.0

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.json
```

Reproduced on 2.5.2 and 2.5.5.

### What happened?

Glob patterns in `plugins[].includes` are matched against the **absolute** file
path, while `files.includes` and `overrides[].includes` are matched relative to
the project root. Because the documented examples and every other `includes`
field take root-relative patterns, a plugin written the same way silently
matches nothing and never reports, with no warning.

In a monorepo whose `biome.json` sits at the repository root, with a plugin
whose pattern is `packages/**/src/**`, the plugin is scoped to zero files. Ours
sat inert for months before we noticed.

Matrix, same config file at the repository root and the same two input files,
changing only `plugins[0].includes`:

| `includes` | diagnostics |
| --- | --- |
| `["packages/**/src/**"]` | 0 |
| `["./packages/**/src/**"]` | 0 |
| `["**/packages/**/src/**"]` | 3 |
| `["**"]` | 3 |
| `["**/src/**"]` | 3 |
| `["/absolute/path/to/repo/packages/**/src/**"]` | 3 |

The absolute pattern working, and the root-relative one not working, is what
points at absolute-path matching.

This is not general glob breakage in the config, which is why it took a while to
diagnose. Two contrasting cases in the same `biome.json`:

- `files.includes` with `!packages/website/src/routeTree.gen.ts` works.
  `biome lint packages/website/src/routeTree.gen.ts` answers `Checked 0 files`
  plus `These paths were provided but ignored`.
- `overrides[].includes` with `packages/app/src/**` works. Disabling a rule
  through that override silenced exactly the diagnostics of that package.

So the same pattern shape is root-relative in `files.includes` and in
`overrides[].includes`, but absolute in `plugins[].includes`.

One consequence worth flagging for anyone hitting this: the negations are inert
too. Anchoring only the positive patterns turns the documented exclusions on for
the first time, which can surface diagnostics in files that were meant to be
excluded. Every negation has to become `!**/...` in the same change.

Reproduction, from a repository root that contains `packages/<name>/src`:

1. Put a GritQL plugin in `biome-plugins/probe.grit` that reports on any
   `export { ... } from '...'`.
2. Register it with `"plugins": [{ "path": "./biome-plugins/probe.grit", "includes": ["packages/**/src/**"] }]`.
3. Run `biome lint packages/<name>/src/<file-with-a-reexport>.ts` and get 0
   plugin diagnostics.
4. Change `includes` to `["**/packages/**/src/**"]` and run the same command to
   get the diagnostics.

### Expected result

`plugins[].includes` should match relative to the project root, the same as
`files.includes` and `overrides[].includes`, so that `packages/**/src/**` scopes
the plugin to that directory.

If absolute matching is intentional, a plugin whose `includes` matches no file
should emit a warning, because the current failure mode is silent: the plugin
loads, compiles, and reports nothing, which is indistinguishable from a clean
codebase.

**Comment by biome-cookie** [comment]
Hello @everton-dgn, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by everton-dgn** [comment]
Minimal reproduction repository: https://github.com/everton-dgn/biome-plugin-includes-glob-repro

It is pinned to Biome 2.5.5. With the root-relative plugin pattern in the repository:

```json
"includes": ["packages/**/src/**"]
```

running:

```sh
pnpm install
pnpm lint
```

produces no plugin diagnostic. Changing only that pattern to:

```json
"includes": ["**/packages/**/src/**"]
```

makes the same command report `The plugin matched this re-export.` on `packages/example/src/index.ts:1`.
